### DISCUSSION TITLE
✅ Should interfaces make ABSTRACT an implicit modifier?

### DISCUSSION THREAD

**Discussion by ZacSweers** [square/kotlinpoet#1051] [General]
I noticed we currently emit `ABSTRACT` modifiers on functions in interfaces, which seems overly redundant. Should we make them implicit?

**Comment by Egorand** [comment]
Unless library mode requires this - totally!

**Reply by Egorand** [reply]
D'oh: #1013, which closes #1011.

I'm not really sure why went with emitting `abstract` on methods over not emitting `abstract` on properties.

**Reply by ZacSweers** [reply]
Yeah I think that's backward. I don't know of this being required with explicit API mode or anything

**Comment by ZacSweers** [comment]
Locking this as resolved ✅
